File: MTGDeckBuilder.Core/DTO/CardSearchCriteria.cs
using MTGDeckBuilder.Core.Domain;

namespace MTGDeckBuilder.Core.DTO
{
    public class CardSearchCriteria
    {
        public IEnumerable<Color> Colors { get; set; }
        public IEnumerable<Format> Formats { get; set; }
        public IEnumerable<Set> Sets { get; set; }
        public IEnumerable<CardType> Types { get; set; }
        public IEnumerable<SuperType> SuperTypes { get; set; }
        public IEnumerable<SubType> SubTypes { get; set; }
        public IEnumerable<Keyword> Keywords { get; set; }
    }
}

File: MTGDeckBuilder.Web/src/app/core/models/card-search-criteria.ts
import { ListItemViewModel } from "./list-item-viewmodel";

export interface CardSearchCriteria {
  colors: ListItemViewModel<number>[];
  sets: ListItemViewModel<number>[];
  formats: ListItemViewModel<number>[];
  types: ListItemViewModel<number>[];
  superTypes: ListItemViewModel<number>[];
  subTypes: ListItemViewModel<number>[];
  keywords: ListItemViewModel<number>[];
}
